### ISSUE TITLE
Logger name repeated twice

### ISSUE THREAD

**Issue by svilen-ivanov** [jOOQ/jOOQ#13906]
### Expected behavior
```
2022-08-23 08:51:15.717 [Test worker]  INFO org.jooq.Constants -  ....
```
The `org.jooq.Constants` correctly reflects the source of the message. 

### Actual behavior
```
2022-08-23 08:51:15.717 [Test worker]  INFO org.jooq.Constants.org.jooq.Constants - 
```
```
2022-08-23 08:51:15.743 [Test worker] DEBUG org.jooq.tools.LoggerListener.org.jooq.tools.LoggerListener - Batch size               : 3
2022-08-23 08:51:15.844 [Test worker] DEBUG org.jooq.tools.LoggerListener.org.jooq.tools.LoggerListener - Executing batch query    : 
```

The location of the log message is duplicated: `org.jooq.Constants.org.jooq.Constants`

### Steps to reproduce the problem
I'm using JOOQ with SLF4J interface (1.7.36) and [Log4J2 Implementation](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl) (2.18.0). I'm using the following lo4j2 pattern:
```xml
<Console name="console" target="SYSTEM_OUT">
    <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %style{[%t]}{magenta} %highlight{%5level}{FATAL=red blink, ERROR=red, WARN=yellow bold, INFO=green, DEBUG=blue, TRACE=normal} %style{%logger}{cyan} - %msg%n"/>
</Console>
```
I think the culprit is this code in `org.jooq.tools.JooqLogger`:

```java
public static JooqLogger getLogger(Class<?> clazz) {
    return getLogger(clazz, clazz.getName(), -1);
}
// ...
public static JooqLogger getLogger(Class<?> clazz, String nameSuffix, int limitMessages) {
    String name = clazz != null && nameSuffix != null
        ? (clazz.getName() + "." + nameSuffix)
        : clazz != null
        ? clazz.getName()
        : nameSuffix;

    JooqLogger result = new JooqLogger(limitMessages);

    // Prioritise slf4j
    try {
        result.slf4j = org.slf4j.LoggerFactory.getLogger(name);
    }
// ...
}
```

that's being called like this:
```java
private static final JooqLogger log        = JooqLogger.getLogger(LoggerListener.class);
```

It is not a big deal, but the double length of the logger name obscures the actual message by making the line unnecessary long.

### Versions

- jOOQ: 3.17.2
- Java: 17 LTS
- Database (include vendor): Postgres 14
- OS: Linux
- JDBC Driver (include name if unofficial driver): Official PG JDBC

**Comment by lukaseder** [comment]
Thanks a lot for your report. Looks like the regression was introduced with a refactoring done for #12681, in jOOQ 3.16

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13908), and 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13907)

**Comment by svilen-ivanov** [comment]
Woah, what I quick response. Thank you @lukaseder
